lib/rack/attack/postrequest.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+module Rack
+  class Attack
+    class Postrequest < Check
+      def initialize(name = nil, &block)
+        super
+        @type = :postrequest
+      end
+
+      def matched_by?(request, response)
```

Comment:
It seems that this method calling twice:
- the first time before `app.call`
- the second time after `app.call`

I guess that we can mark request as matched even when it doesn't blocked.

Reply:
mb change `postrequest.matched_by?` to `postrequest.block.call`?
 
#lib/rack/attack/configuration.rb
```
def process_postrequests(request, response)
    @postrequests.each { |_name, postrequest| postrequest.matched_by?(request, response) }
end
```